### PR TITLE
move the library LWRPs to custom resources

### PR DESCRIPTION
**Pull request by lamont-granquist**
also updates for modern foodcritic and fauxhai

### CODE REVIEW

File: metadata.rb [hunk 1]
```diff
@@ -10,3 +10,9 @@
 
 source_url 'https://github.com/chef-cookbooks/enterprise' if respond_to?(:source_url)
```
Comment (tas50) [inline metadata.rb]: respond_tos can go away

File: resources/component_runit_supervisor_upstart.rb [hunk 2]
```diff
@@ -0,0 +1,67 @@
+include ComponentRunitSupervisorResourceMixin
+
+provides :component_runit_supervisor, platform_family: 'rhel' do |node|
+  node['platform_version'].to_i == 6
+end
+provides :component_runit_supervisor, platform: 'fedora' do |node|
+  node['platform_version'].to_i <= 14
```
Comment (tas50) [inline resources/component_runit_supervisor_upstart.rb]: We can kill this. If you're running Fedora 14 you should get a horrible error. The thing went EOL years ago

Comment (lamont-granquist) [inline resources/component_runit_supervisor_upstart.rb]: i'm aggressively not giving a shit about those provides lines and just copying what was there, i think its all bad, but i don't have time to rethink it all right now.

**Comment by tas50** [comment]
Ship it. Also that FC016 mess goes away once we get a new ChefDK out

**Comment by lamont-granquist** [comment]
@tas50 this is also probably how you should clean up the mysql+httpd resources+providers if you haven't already, and is more or less the pattern to follow to deal with code duplication in custom resources.

**Pull request merged**